### ISSUE TITLE
node.push() fails silently if the node has an empty list in it's properties

### ISSUE THREAD

**Issue by LynnCo** [nigelsmall/py2neo#395]
If you get a currently existing node in the database, and then do:

    n = db.find_one('some_label')
    n['empty_list'] = []
    n.push()
    n.pull()
    print(n['empty_list'])

You'll get None because n.push() fails silently when there's an empty list on the properties. Some things to note:

* the entire push fails, not just pushing that single attribute
* there's no issue accessing the attribute from node.properties
* it's only empty lists that do this. Empty strings and 0 are both fine

Oh, also this is with py2neo 2.0.6 and python 3.4.2

**Comment by nigelsmall** [comment]
Thanks, looks like a bug. Checks are made to ascertain that list properties are homogenous and I've probably missed the empty list case here.

I'll make a fix for the next point release.

**Comment by nigelsmall** [comment]
This is addressed in 370dbf0 and turns out to be a server limitation of which I was previously unaware. Empty lists cannot be added as property values and an exception is generally thrown if this is attempted:

```
PropertyValueException: Unable to set property 'faults' to an empty array, because,
since there are no values of any type in it, and no pre-existing collection to infer
type from, it is not possible to determine what type of array to store.
```

The previous implementation hid this error by not parsing the batch response.
